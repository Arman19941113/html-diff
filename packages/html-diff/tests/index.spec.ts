import { describe, expect, it } from 'vitest'
import HtmlDiff from '../src'

describe('HtmlDiff', () => {
  it('should work with basic create', function () {
    const oldHtml = `<div>hello</div>`
    const newHtml = `<div>hello world</div>`
    const diff = new HtmlDiff(oldHtml, newHtml)
    expect(diff.getUnifiedContent()).toMatchSnapshot()
    expect(diff.getSideBySideContents()).toMatchSnapshot()
  })

  it('should work with basic delete', function () {
    const oldHtml = `<div>hello world</div>`
    const newHtml = `<div>hello</div>`
    const diff = new HtmlDiff(oldHtml, newHtml)
    expect(diff.getUnifiedContent()).toMatchSnapshot()
    expect(diff.getSideBySideContents()).toMatchSnapshot()
  })

  it('should work with basic replace', function () {
    const oldHtml = `<div>hello</div>`
    const newHtml = `<div>world</div>`
    const diff = new HtmlDiff(oldHtml, newHtml)
    expect(diff.getUnifiedContent()).toMatchSnapshot()
    expect(diff.getSideBySideContents()).toMatchSnapshot()
  })

  it('should work sample 1', function () {
    const oldHtml = `<div>hello world</div>`
    const newHtml = `<h1>You got a dream. You gotta protect it.</h1>`
    const diff = new HtmlDiff(oldHtml, newHtml)
    expect(diff.getUnifiedContent()).toMatchSnapshot()
    expect(diff.getSideBySideContents()).toMatchSnapshot()
  })

  it('should work sample 2', function () {
    const oldHtml = `<h1>Hello</h1>
<h2>Let life be beautiful like summer flower and death like autumn leaves.</h2>
<p>She could fade and wither- I didn't care. I would still go mad with tenderness at the mere sight of her face.</p>
<p>她可以褪色，可以枯萎，怎样都可以。但只要我看她一眼，万般柔情便涌上心头。</p>
<p>夜已深 我心思思 你的丰姿</p>
<p>只想你便是 我的天使</p>
<p>未见半秒 便控制不了</p>
<p>难以心安 于今晚</p>
<img src="./dog.jpg" alt="dog"><h2>Try video</h2>
<video style="width: 100%;" controls><source src="https://www.w3schools.com/html/mov_bbb.mp4" type="video/mp4"></video>`
    const newHtml = `<h1>Hello World</h1><p>She could fade and wither. I would still go mad with tenderness at the mere sight of her face.</p>
<p>她可以褪色，可以枯萎。但只要我看她一眼，万般柔情便涌上了我的心头。</p>
<p>让我靠着你的臂胳</p>
<p>流露我热爱心底说话</p>
<p>孕育美丽温馨爱意</p>
<p>做梦 都是你</p>
<img src="./cat.jpg" alt="cat"><h2>Try video</h2>
<h2>Set the bird's wings with gold and it will never again soar in the sky.</h2>
<video style="width: 100%;" controls src="https://www.w3schools.com/html/mov_bbb.mp4" type="video/mp4"></video>`

    const diff = new HtmlDiff(oldHtml, newHtml, 3)
    expect(diff.getUnifiedContent()).toMatchSnapshot()
    expect(diff.getSideBySideContents()).toMatchSnapshot()
  })

  it('should work sample 3', function () {
    const oldHtml = `<h1>hello world</h1>
<p>勤字功夫，第一贵早起，第二贵有恒。</p>
<p>流水不争先，争的是滔滔不绝</p>
<p>即便再痛苦，也不要选择放弃！</p>
<p>不相信自己的人，连努力的价值都没有！</p>
<p>今天和明天已经由昨天决定，你还可以决定后天。</p>
<p>只有当你离开自己的舒适区时，你才会挑战自己的极限。</p>
<p>一本有价值的书就是一盏智慧之灯，总有人不断从中提取光明。</p>`
    const newHtml = `<h1>hello world</h1>
<p>勤字功夫，第一贵早起，第二贵有恒。</p>
<p>流水不争先，争的是滔滔不绝</p>
<p>痛苦，要选择放弃！</p>
<p>不相信自己的人，也有努力的价值！</p>
<p>无休止的欲望像个黑洞，浸染了我们原本澄澈而简单的心</p>
<p>只有当你离开自己的舒适区时，你才会挑战自己的极限。</p>
<p>一本有价值的书就是一盏智慧之灯，总有人不断从中提取光明。</p>`

    const diff = new HtmlDiff(oldHtml, newHtml)
    expect(diff.getUnifiedContent()).toMatchSnapshot()
    expect(diff.getSideBySideContents()).toMatchSnapshot()
  })
})
