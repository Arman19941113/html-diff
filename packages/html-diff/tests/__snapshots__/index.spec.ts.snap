// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HtmlDiff > should work sample 1 1`] = `"<div><span class="html-diff-delete-text-wrapper">hello world</span></div><h1><span class="html-diff-create-text-wrapper">You got a dream. You gotta protect it.</span></h1>"`;

exports[`HtmlDiff > should work sample 1 2`] = `
[
  "<div><span class="html-diff-delete-text-wrapper">hello world</span></div>",
  "<h1><span class="html-diff-create-text-wrapper">You got a dream. You gotta protect it.</span></h1>",
]
`;

exports[`HtmlDiff > should work sample 2 1`] = `
"<h1><span class="del-txt">Hello</span><span class="cra-txt">Hello World</span></h1><h2><span class="del-txt">Let life be beautiful like summer flower and death like autumn leaves.</span></h2><p>She could fade and wither<span class="del-txt">- I didn't care</span>. I would still go mad with tenderness at the mere sight of her face.</p>
<p>她可以褪色，可以枯萎<span class="del-txt">，怎样都可以</span>。但只要我看她一眼，万般柔情便涌上<span class="cra-txt">了我的</span>心头。</p>
<p><span class="del-txt">夜已深 我心思思 你的丰姿</span><span class="cra-txt">让我靠着你的臂胳</span></p>
<p><span class="del-txt">只想你便是 我的天使</span><span class="cra-txt">流露我热爱心底说话</span></p>
<p><span class="del-txt">未见半秒 便控制不了</span><span class="cra-txt">孕育美丽温馨爱意</span></p>
<p><span class="del-txt">难以心安 于今晚</span><span class="cra-txt">做梦 都是你</span></p><span class="del-inl"><img src="./dog.jpg" alt="dog"></span><span class="cra-inl"><img src="./cat.jpg" alt="cat"></span><h2>Try video</h2>
<div class="del-blo"><video style="width: 100%;" controls><source src="https://www.w3schools.com/html/mov_bbb.mp4" type="video/mp4"></video></div><h2><span class="cra-txt">Set the bird's wings with gold and it will never again soar in the sky.</span></h2><div class="cra-blo"><video style="width: 100%;" controls src="https://www.w3schools.com/html/mov_bbb.mp4" type="video/mp4"></video></div>"
`;

exports[`HtmlDiff > should work sample 2 2`] = `
[
  "<h1><span class="del-txt">Hello</span></h1><h2><span class="del-txt">Let life be beautiful like summer flower and death like autumn leaves.</span></h2><p data-seq="1">She could fade and wither<span class="del-txt">- I didn't care</span>. I would still go mad with tenderness at the mere sight of her face.</p>
<p data-seq="2">她可以褪色，可以枯萎<span class="del-txt">，怎样都可以</span>。但只要我看她一眼，万般柔情便涌上心头。</p>
<p data-seq="3"><span class="del-txt">夜已深 我心思思 你的丰姿</span></p>
<p data-seq="4"><span class="del-txt">只想你便是 我的天使</span></p>
<p data-seq="5"><span class="del-txt">未见半秒 便控制不了</span></p>
<p data-seq="6"><span class="del-txt">难以心安 于今晚</span></p><span class="del-inl"><img src="./dog.jpg" alt="dog"></span><h2 data-seq="7">Try video</h2>
<div class="del-blo"><video style="width: 100%;" controls><source src="https://www.w3schools.com/html/mov_bbb.mp4" type="video/mp4"></video></div>",
  "<h1><span class="cra-txt">Hello World</span></h1><p data-seq="1">She could fade and wither. I would still go mad with tenderness at the mere sight of her face.</p>
<p data-seq="2">她可以褪色，可以枯萎。但只要我看她一眼，万般柔情便涌上<span class="cra-txt">了我的</span>心头。</p>
<p data-seq="3"><span class="cra-txt">让我靠着你的臂胳</span></p>
<p data-seq="4"><span class="cra-txt">流露我热爱心底说话</span></p>
<p data-seq="5"><span class="cra-txt">孕育美丽温馨爱意</span></p>
<p data-seq="6"><span class="cra-txt">做梦 都是你</span></p><span class="cra-inl"><img src="./cat.jpg" alt="cat"></span><h2 data-seq="7">Try video</h2>
<h2><span class="cra-txt">Set the bird's wings with gold and it will never again soar in the sky.</span></h2><div class="cra-blo"><video style="width: 100%;" controls src="https://www.w3schools.com/html/mov_bbb.mp4" type="video/mp4"></video></div>",
]
`;

exports[`HtmlDiff > should work sample 3 1`] = `
"<h1>hello world</h1>
<p>勤字功夫，第一贵早起，第二贵有恒。</p>
<p>流水不争先，争的是滔滔不绝</p>
<p><span class="del-txt">即便再</span>痛苦，<span class="del-txt">也不</span>要选择放弃！</p>
<p>不相信自己的人，<span class="del-txt">连</span><span class="cra-txt">也有</span>努力的价值<span class="del-txt">都没有</span>！</p>
<p><span class="del-txt">今天和明天已经由昨天决定，你还可以决定后天。</span><span class="cra-txt">无休止的欲望像个黑洞，浸染了我们原本澄澈而简单的心</span></p>
<p>只有当你离开自己的舒适区时，你才会挑战自己的极限。</p>
<p>一本有价值的书就是一盏智慧之灯，总有人不断从中提取光明。</p>"
`;

exports[`HtmlDiff > should work sample 3 2`] = `
[
  "<h1 data-seq="1">hello world</h1>
<p data-seq="2">勤字功夫，第一贵早起，第二贵有恒。</p>
<p data-seq="3">流水不争先，争的是滔滔不绝</p>
<p data-seq="4"><span class="del-txt">即便再</span>痛苦，<span class="del-txt">也不</span>要选择放弃！</p>
<p data-seq="5">不相信自己的人，<span class="del-txt">连</span>努力的价值<span class="del-txt">都没有</span>！</p>
<p data-seq="6"><span class="del-txt">今天和明天已经由昨天决定，你还可以决定后天。</span></p>
<p data-seq="7">只有当你离开自己的舒适区时，你才会挑战自己的极限。</p>
<p data-seq="8">一本有价值的书就是一盏智慧之灯，总有人不断从中提取光明。</p>",
  "<h1 data-seq="1">hello world</h1>
<p data-seq="2">勤字功夫，第一贵早起，第二贵有恒。</p>
<p data-seq="3">流水不争先，争的是滔滔不绝</p>
<p data-seq="4">痛苦，要选择放弃！</p>
<p data-seq="5">不相信自己的人，<span class="cra-txt">也有</span>努力的价值！</p>
<p data-seq="6"><span class="cra-txt">无休止的欲望像个黑洞，浸染了我们原本澄澈而简单的心</span></p>
<p data-seq="7">只有当你离开自己的舒适区时，你才会挑战自己的极限。</p>
<p data-seq="8">一本有价值的书就是一盏智慧之灯，总有人不断从中提取光明。</p>",
]
`;

exports[`HtmlDiff > should work with basic create 1`] = `"<div>hello<span class="html-diff-create-text-wrapper"> world</span></div>"`;

exports[`HtmlDiff > should work with basic create 2`] = `
[
  "<div data-seq="1">hello</div>",
  "<div data-seq="1">hello<span class="html-diff-create-text-wrapper"> world</span></div>",
]
`;

exports[`HtmlDiff > should work with basic delete 1`] = `"<div>hello<span class="html-diff-delete-text-wrapper"> world</span></div>"`;

exports[`HtmlDiff > should work with basic delete 2`] = `
[
  "<div data-seq="1">hello<span class="html-diff-delete-text-wrapper"> world</span></div>",
  "<div data-seq="1">hello</div>",
]
`;

exports[`HtmlDiff > should work with basic replace 1`] = `"<div><span class="html-diff-delete-text-wrapper">hello</span><span class="html-diff-create-text-wrapper">world</span></div>"`;

exports[`HtmlDiff > should work with basic replace 2`] = `
[
  "<div><span class="html-diff-delete-text-wrapper">hello</span></div>",
  "<div><span class="html-diff-create-text-wrapper">world</span></div>",
]
`;

exports[`HtmlDiff > should work with equal 1`] = `
"<h1>hello world</h1>
<p>你若安好，便是晴天</p>"
`;

exports[`HtmlDiff > should work with equal 2`] = `
[
  "<h1 data-seq="1">hello world</h1>
<p data-seq="2">你若安好，便是晴天</p>",
  "<h1 data-seq="1">hello world</h1>
<p data-seq="2">你若安好，便是晴天</p>",
]
`;

exports[`HtmlDiff > should work with equal double 1`] = `
"<p>hello world</p>
<p><span class="html-diff-delete-text-wrapper">你若安好，便是晴天</span><span class="html-diff-create-text-wrapper">今天天气很不错</span></p>
<p>你的微笑总是让我为你着迷</p>"
`;

exports[`HtmlDiff > should work with equal double 2`] = `
[
  "<p data-seq="1">hello world</p>
<p data-seq="2"><span class="html-diff-delete-text-wrapper">你若安好，便是晴天</span></p>
<p data-seq="3">你的微笑总是让我为你着迷</p>",
  "<p data-seq="1">hello world</p>
<p data-seq="2"><span class="html-diff-create-text-wrapper">今天天气很不错</span></p>
<p data-seq="3">你的微笑总是让我为你着迷</p>",
]
`;

exports[`HtmlDiff > should work with equal end 1`] = `
"<p>你<span class="html-diff-delete-text-wrapper">有一双会说话的眼睛</span><span class="html-diff-create-text-wrapper">的微笑总是让我为你着迷</span></p>
<p>你若安好，便是晴天</p>"
`;

exports[`HtmlDiff > should work with equal end 2`] = `
[
  "<p data-seq="1">你<span class="html-diff-delete-text-wrapper">有一双会说话的眼睛</span></p>
<p data-seq="2">你若安好，便是晴天</p>",
  "<p data-seq="1">你<span class="html-diff-create-text-wrapper">的微笑总是让我为你着迷</span></p>
<p data-seq="2">你若安好，便是晴天</p>",
]
`;

exports[`HtmlDiff > should work with equal start 1`] = `
"<p>hello world</p>
<p><span class="html-diff-delete-text-wrapper">你若安好，便是晴天</span><span class="html-diff-create-text-wrapper">今天天气很不错</span></p>"
`;

exports[`HtmlDiff > should work with equal start 2`] = `
[
  "<p data-seq="1">hello world</p>
<p data-seq="2"><span class="html-diff-delete-text-wrapper">你若安好，便是晴天</span></p>",
  "<p data-seq="1">hello world</p>
<p data-seq="2"><span class="html-diff-create-text-wrapper">今天天气很不错</span></p>",
]
`;

exports[`HtmlDiff > should work with white space 1`] = `"<div>hello</div>"`;

exports[`HtmlDiff > should work with white space 2`] = `
[
  "<div data-seq="1">hello</div>",
  "<div data-seq="1">hello</div>",
]
`;
